ROADMAP.md

```diff
@@ -1,10 +1,101 @@
+* v2 python components.
+  * A convenient component authoring method designed to support new features above natively. (v1 components do not completely support all the features mentioned above)
+  * [samples/test/lightweight_python_functions_v2_with_outputs.py](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_with_outputs.py)
+  * [samples/test/lightweight_python_functions_v2_pipeline.py](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_pipeline.py)
+* [#5669](https://github.com/kubeflow/pipelines/issues/5669) KFP semantics in MLMD
+  * MLMD state with exact KFP semantics (e.g. parameter / artifact / task names are the same as in DSL). This will enable use-cases like: “querying a result artifact from a pipeline run using MLMD API and then use the result in another system or another pipeline”.
+  
+    [Example pipeline](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_pipeline.py) and [corresponding MLMD state](https://github.com/kubeflow/pipelines/blob/master/samples/test/lightweight_python_functions_v2_pipeline_test.py).
+* [#5670](https://github.com/kubeflow/pipelines/issues/5670) Revamp KFP UI to show inputs & outputs in KFP semantics for v2 compatible pipelines
+* [#5667](https://github.com/kubeflow/pipelines/issues/5667) KFP data model based caching using MLMD.
+
+### KFP v2
+
+Design: [bit.ly/kfp-v2](https://bit.ly/kfp-v2)
+
+#### KFP v2 Goals
+
+* Data Management:
+  * build first class support for metadata -- recording, presentation and orchestration.
+  * making it easy to keep track of all the data produced by machine learning pipelines and how it was computed.
+* KFP native (and argo agnostic) spec and status: define a clear interface for KFP, so that other systems can understand KFP pipeline spec and status in KFP semantics.
+* Gaining more control over KFP runtime behavior: give the KFP system more control over the exact runtime behavior. This wasn’t a goal initially coming from use cases. However, the more we innovate on Data Management and KFP native spec/status, the clearer that other workflow systems become limitations to how we may implement new KFP features on our own. Therefore, re-architecturing KFP to let us get more control of runtime behavior is ideal for achieving KFP’s long term goals.
+* Be backward compatible with KFP v1: for existing features, we want to keep them as backward compatible as possible to ease upgrade.
+
+#### Timeline - v2
+
+* Start work after v2 compatible feature complete
+* Alpha release in October
+* Beta/Stable release timelines TBD
+
+#### Planned Features in KFP v2
+
+* KFP v2 DSL (TBD)
+
+* Use [the pipeline spec](https://github.com/kubeflow/pipelines/blob/master/api/v2alpha1/pipeline_spec.proto) as pipeline description data structure. The new spec is argo workflow agnostic and can be a shared common format for different underlying engines.
+
+* KFP v2 DAG UI
```

Comment:
Do we also mention storing and reading pipeline run status to MLMD?

Reply:
that feels like an implementation detail to me.
I included that in user facing item: "KFP UI in KFP semantics".

Reply:
Sounds good to me, thank you!